includes/Tasks/FetchUrlsRecursiveTask.php

```diff
@@ -0,0 +1,135 @@
+use WP_Error;
+
+/**
+ * Class which handles fetch recursive urls task; all urls beneath the current post_id.
+ */
+class FetchUrlsRecursiveTask extends Fetch_Urls_Task
+{
+    /**
+     * Task name.
+     *
+     * @var string
+     */
+    public static $task_name = 'fetch_urls_recursive';
+
+    /**
+     * Fetch and save pages for the static archive
+     *
+     * @return boolean|WP_Error true if done, false if not done, WP_Error if error.
+     */
+    public function perform()
+    {
+        $batch_size = apply_filters('simply_static_fetch_urls_batch_size', 10);
+        $post_id = get_option(Plugin::SLUG . '_single_deploy_id');
+        $post_url = get_permalink($post_id);
+
+        $this->save_status_message(
+            "Fetching urls recursively from post with post id: ${post_id} and url: ${post_url}."
+        );
+
+        $static_pages = apply_filters(
+            'ss_static_pages',
+            Page::query()
+                ->where('last_checked_at < ? OR last_checked_at IS NULL', $this->archive_start_time)
+                ->limit($batch_size)
+                ->find(),
+            $this->archive_start_time
+        );
+
+        $pages_remaining = apply_filters(
+            'ss_remaining_pages',
+            Page::query()
+                ->where('last_checked_at < ? OR last_checked_at IS NULL', $this->archive_start_time)
+                ->count(),
+            $this->archive_start_time
+        );
+
+        $total_pages = apply_filters('ss_total_pages', Page::query()->count());
+
+        $pages_processed = $total_pages - $pages_remaining;
+        Util::debug_log("Total pages: " . $total_pages . '; Pages remaining: ' . $pages_remaining);
+
+        while ($static_page = array_shift($static_pages)) {
+            Util::debug_log("URL: " . $static_page->url);
+
+            $excludable = $this->find_excludable($static_page);
+            if ($excludable !== false) {
+                $save_file = $excludable['do_not_save'] !== '1';
+                $follow_urls = $excludable['do_not_follow'] !== '1';
+
+                Util::debug_log("Excludable found: URL: " . $excludable['url'] . ' DNS: ' . $excludable['do_not_save'] . ' DNF: ' . $excludable['do_not_follow']);
+            } else {
+                $save_file = true;
+                $follow_urls = true;
+                Util::debug_log("URL is not being excluded");
+            }
+
+            // If we're not saving a copy of the page or following URLs on that
+            // page, then we don't need to bother fetching it.
+            if ($save_file === false && $follow_urls === false) {
+                Util::debug_log("Skipping URL because it is no-save and no-follow");
+                $static_page->last_checked_at = Util::formatted_datetime();
+                $static_page->set_status_message(__("Do not save or follow", 'simply-static'));
+                $static_page->save();
+                continue;
+            }
+
+            $url = $static_page->url;
```

Comment:
Volgens mij komen we hier op het punt dat we onze eigen custom code gebruiken Misschien nog fijn om dat in een comment aan te geven.